### PR TITLE
test-lib: add functions to test image availability

### PR DESCRIPTION
**Pull request by pkubatrh**
test-lib: change default rhel7/rhel8 registry to registry.redhat.io

### CODE REVIEW

**Comment by pkubatrh** [comment]
[test]

@phracek PTAL

File: test-lib.sh [hunk 1]
```diff
@@ -430,6 +430,32 @@ ct_registry_from_os() {
...
+  local public_image_name
+  local registry
+
+  registry=$(ct_registry_from_os "$os")
+  if [ "x$os" == "xrhel7" ]; then
+    public_image_name=$registry/rhscl/$base_image_name-${version//./}-rhel7
+  elif [ "x$os" == "xrhel8" ]; then
+    public_image_name=$registry/rhel8/$base_image_name-${version//./}
+  elif [ "x$os" == "xcentos7" ]; then
+    public_image_name=$registry/centos/$base_image_name-${version//./}-centos7
```
Comment (phracek) [inline test-lib.sh]: Fedora is missing, If it is relevant.

File: test-lib.sh [hunk 2]
```diff
@@ -430,6 +430,32 @@ ct_registry_from_os() {
   echo "$registry"
 }
 
+ # ct_get_public_image_name OS BASE_IMAGE_NAME VERSION
+# ----------------
+# Transform the arguments into public image name
+# Argument: OS - string containing the os version
+# Argument: BASE_IMAGE_NAME - string containing the base name of the image as defined in the Makefile
+# Argument: VERSION - string containing the version of the image as defined in the Makefile
+ct_get_public_image_name() {
```
Comment (phracek) [inline test-lib.sh]: And of course. It is a good candidate to write a simple test for each `OS`.
Only this is missing and test is simple.

Please the test here https://github.com/sclorg/container-common-scripts/tree/master/tests/test-lib and call it from 
https://github.com/sclorg/container-common-scripts/blob/master/Makefile#L15

the test can be called `get_public_image_name`.

File: tests/test-lib/image_availability [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -e
+
+. test-lib.sh
+
+# This should succeed
+if ! ct_check_image_availability docker.io/centos/postgresql-10-centos7; then
+  echo "image_availabily test failed"
```
Comment (phracek) [inline tests/test-lib/image_availability]:
```suggestion
  echo "image_availability test failed"
```

File: tests/test-lib/image_availability [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+fi
+
+# This should fail
+if ct_check_image_availability docker.io/centos/postgresql-98-centos7; then
+  echo "image_availabily test failed"
```
Comment (phracek) [inline tests/test-lib/image_availability]:
```suggestion
  echo "image_availability test failed"
```

File: tests/test-lib/image_availability [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+fi
+
+echo "image_availabily test completed successfully."
```
Comment (phracek) [inline tests/test-lib/image_availability]:
```suggestion
echo "image_availability test completed successfully."
```

**Comment by pkubatrh** [comment]
@phracek I added some simple tests for both functions, PTAL

**Comment by pkubatrh** [comment]
[test]

**Comment by phracek** [comment]
Thanks for the tests.

File: test-lib.sh (hunk 1 above)
Comment (pkubatrh) [reply test-lib.sh]: Fedora public image was not used before and it is slightly harder to create the correct one with fedora versions changing often, so I left it out for now. Can be easily added later if needed.

**Comment by pkubatrh** [comment]
Fixed typos

**Comment by pkubatrh** [comment]
[test]

**Pull request merged**